crates/bevy_render/src/mesh/mesh/conversions.rs

```diff
@@ -25,9 +25,21 @@
 //! ```
 
 use crate::mesh::VertexAttributeValues;
+use bevy_math::{IVec2, IVec3, IVec4, UVec2, UVec3, UVec4, Vec2, Vec3, Vec4};
 use bevy_utils::EnumVariantMeta;
 use thiserror::Error;
 
+macro_rules! convert_vec_of_vec {
+    ($x:expr, $t:ty, $s:literal) => {
+        unsafe {
+            // SAFETY: x is only used in this file. It is called only with x being a value of type `Vec<Vec3>` or something similar.
+            // Vec3 is a repr(C) struct containing 3 f32s, hence it is transmutable to a [f32; 3].
```

Comment:
Where in the glam docs are we given this guarantee? Doing a search on the page for Vec3 and across the entire glam docs for `Layout` gives no result.

Reply:
In relevant part:
```rust
#[derive(Clone, Copy, Debug, Default, PartialEq, PartialOrd)]
#[cfg_attr(target_arch = "spirv", repr(simd))]
#[cfg_attr(not(target_arch = "spirv"), repr(C))]
pub struct XYZ<T> {
    pub x: T,
    pub y: T,
    pub z: T,
}

#[repr(transparent)]
pub struct Vec3(pub(crate) XYZF32);
```

The second is `repr(transparent)` and the former is `repr(C)` so overall `Vec3` is `repr(C)`.

Reply:
Also note that while I don't have the reference right now, I found some docs stating that `repr(simd)` is transmutable to an array. Will get the reference tomorrow.

Reply:
That doesn't answer the question - I asked where glam gives us this guarantee. You have responded showing how the current implementation has this property

Those are not the same thing.

Reply:
A choice of `repr` is inherently part of the API for exactly this reason. By choosing `repr(C)` you are making the promise that your code can interop with C, so removing it is an API break. Another way to see this is to note that the `repr` appears in rustdoc.

Reply:
From the nomicon (https://doc.rust-lang.org/nomicon/other-reprs.html#reprtransparent):
> This repr is only considered part of the public ABI of a type if either the single field is pub, or if its layout is documented in prose. Otherwise, the layout should not be relied upon by other crates.

-> Vec3 uses only `pub(crate)`
-> should not be relied upon.

Reply:
Alright, thanks. If I got a PR on glam merged to add docs for this would that be sufficient to unblock this?